src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -70,12 +70,38 @@ class OperationalCredentialsAttrAccess : public AttributeAccessInterface
     CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
 
 private:
+    CHIP_ERROR ReadNOCs(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadSupportedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadCommissionedFabrics(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadFabricsList(EndpointId endpoint, AttributeValueEncoder & aEncoder);
     CHIP_ERROR ReadRootCertificates(EndpointId endpoint, AttributeValueEncoder & aEncoder);
 };
 
+CHIP_ERROR OperationalCredentialsAttrAccess::ReadNOCs(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    auto accessingFabricIndex = aEncoder.AccessingFabricIndex();
+
+    return aEncoder.EncodeList([accessingFabricIndex](const auto & encoder) -> CHIP_ERROR {
+        for (auto & fabricInfo : Server::GetInstance().GetFabricTable())
+        {
+            if (!fabricInfo.IsInitialized())
+                continue;
+
+            if (accessingFabricIndex == fabricInfo.GetFabricIndex())
+            {
+                Clusters::OperationalCredentials::Structs::NOCStruct::Type noc;
+
+                noc.fabricIndex = fabricInfo.GetFabricIndex();
```

Comment:
I believe per spec all the structs get encoded.  It's just specific fields that get redacted from the structs that don't match the accessing fabric.